### ISSUE TITLE
path selector for the cache

### ISSUE THREAD

**Issue by KonstantinDjairo** [xiaoyifang/goldendict-ng#1155]
i've noticed that goldendict-ng consumes a lot of storage when indexing dictionaries, sometimes more than 40gb.
here i selected just a few dictionaries, so it won't consume a lot of storage from the root partition , but still having 23gb of cache
![image](https://github.com/xiaoyifang/goldendict-ng/assets/53496273/38e24a22-3d97-49e1-bc98-43f9651f48a0)
it'd be really nice if we had an option to select the path where we want to store the cache, some people really have lots of dictionaries and sometimes there isn't enough space in the root partition. But with that change the user could simply select another folder located in a separated HDD for example.

**Comment by xiaoyifang** [comment]
> it'd be really nice if we had an option to select the path where we want to store the cache,

This is a good idea.

For now ,I think maybe you can use linked directory?

**Comment by KonstantinDjairo** [comment]
yeah, by now I'm using this, but there are many downsides , since I'm linking the cache folder entirely , the whole OS gets slow.
With the proposed modification this issue might not occur.
